File: backend/src/main/java/org/example/backend/global/exception/ServiceException.java
package org.example.backend.global.exception;

import lombok.Getter;
import org.example.backend.global.response.ApiResponse;
import org.springframework.http.HttpStatus;

@Getter
public class ServiceException extends RuntimeException {

    private final String resultCode;
    private final String msg;
    private final HttpStatus httpStatus;

    public ServiceException(String resultCode, String msg, HttpStatus httpStatus) {
        super("%s : %s".formatted(resultCode, msg));
        this.resultCode = resultCode;
        this.msg = msg;
        this.httpStatus = httpStatus;
    }
    public ApiResponse<Void> getApiResponse() {
        return new ApiResponse<>(resultCode, msg);
    }
}

File: backend/src/main/java/org/example/backend/global/exception/ServiceException.kt
package org.example.backend.global.exception

import org.example.backend.global.response.ApiResponse
import org.springframework.http.HttpStatus

class ServiceException(
    val resultCode: String,
    val msg: String,
    val httpStatus: HttpStatus
) : RuntimeException(
    "$resultCode : $msg"
) {
    fun getApiResponse(): ApiResponse<Void?> {
        return ApiResponse(resultCode, msg)
    }
}
